`azurerm_virtual_network_gateway`: bgp setting support `APIPA` (#10381)
terraform-providers/terraform-provider-azurerm: website/docs/r/virtual_network_gateway.html.markdown

    The supported values are `SSTP`, `IkeV2` and `OpenVPN`.
    Values `SSTP` and `IkeV2` are incompatible with the use of 
    `aad_tenant`, `aad_audience` and `aad_issuer`.
  
---

The `bgp_settings` block supports:

* `asn` - (Optional) The Autonomous System Number (ASN) to use as part of the BGP.

> * `peering_addresses` - (Optional) A list of `peering_addresses` as defined below. You can specify either one `peering_addresses` (normal use case) or two (when `active_active` is `true`).

Reviewer: grammer
```suggestion
* `peering_addresses` - (Optional) A list of `peering_addresses` as defined below. Only one `peering_addresses` block can be specified except when `active_active` on the ???? is `true`.
```

* `peer_weight` - (Optional) The weight added to routes which have been learned
    through BGP peering. Valid values can be between `0` and `100`.
  
---

A `custom_route` block supports the following:

* `address_prefixes` - (Optional) A list of address blocks reserved for this virtual network in CIDR notation. Changing this forces a new resource to be created.

---

A `peering_addresses` supports the following:

> * `ip_configuration_name` - (Required) The name of the IP configuration of this Virtual Network Gateway (as you defined in the`ip_configuration`).

Reviewer: if this is defined in the `ip_configuration` for the gateway couldn't we just pull it from that property?

Author: In case there are multiple `peering_addresses`(s), in which case there are also multiple `ip_configuration`(s), you need to specify which `ip_configuration_name` this setting correspond to.

Reviewer: if there is 1 and 1 we should be able to infer this and error out of we can't?

* `apipa_addresses` - (Optional) A list of Azure custom APIPA addresses assigned to the BGP peer of the Virtual Network Gateway.
  

> ~> **Note:** The valid range for the reserved APIPA address in Azure Public is from "169.254.21.0" to "169.254.22.255".

Reviewer: could we correctly quote these values?
```suggestion
~> **Note:** The valid range for the reserved APIPA address in Azure Public is from `169.254.21.0` to `169.254.22.255`.
```

    The supported values are `SSTP`, `IkeV2` and `OpenVPN`.
    Values `SSTP` and `IkeV2` are incompatible with the use of 
    `aad_tenant`, `aad_audience` and `aad_issuer`.
  
---

The `bgp_settings` block supports:

* `asn` - (Optional) The Autonomous System Number (ASN) to use as part of the BGP.

> * `peering_address` - (Optional / **Deprecated**) The BGP peer IP address of the virtual network

Author: Yes, I'll remove the `ForceNew` option and remove this line also.
